### ISSUE TITLE
[css-page-floats] inline alignment of block-end floats

### ISSUE THREAD

**Issue by frivoal** [w3c/csswg-drafts#1236]
[the definition of `float: block-end`](https://drafts.csswg.org/css-page-floats/#valdef-float-block-end) calls for
> a box that is floated to the block-end and line-end outer edges of the float reference.

This is what Vivliostyle implements, but is different from what prince and antena house do: they float to block-end and **line-start**.

Later version of the spec should definitely make this an author selectable choice (e.g. `float: bottom right`), but until then, we think it may be preferable to align on these existing implementations.

**Comment by frivoal** [comment]
[live example](https://kwkbtr.github.io/vivliostyle-live-viewer/#7VfLjttGELz7KzoKctOSiY08IHN1iO2c8tjAGwM5NmdaVCPDGe48uDaC/HtqSGkV7x8E0ElUd093dVUNIXVfvP3tzf2fd+/omEe3f9HVD3Lsh9uN+E0NCNv9C6JulMxkjhyT5NvNH/c/3fywWRJZs5P9HQ9CBxc4J9KUitCX37x89V3XrulamPKn9YmoyWHaUtOHnMNIfy8xoj5EK3FHvcPxl9NHSsGpfX3KjhwH9Tv6dvp4Dk1srfrhs9iC4SbKQaJ4IzsUDXJOPqrNxx19//VX58hRdDhmdJBxDf3zBPAJ19JyRwh9VvIM/alqjV4Ku/a0d9euXHZ9sJ8WQqzOZByndLtB783+HjOXLl2L1POStfFm/+M69lnhtH/3kcfJQYQQqWM6goLbzTHnKe3adtB8LH1jwtg+vjKtSelxuLGRDzm1i1yprXJt9otqO1Lv1Aux08GP4jOFA2QJ5q8b8fakc9fyvmun0/ifQ5SRdEplJBscQCTNxLDNlkzwSUyWXCKx1UmTgWwkTnNDvxdNDxDcsSmJJKEej+qV6QEpckvjWTNLzeCbZEpGnMT1oC/OcUNvgxdDaiGptxHJTBJDauh9SRNCWBMLFZMrNHRkQyF6jlLhzeycovYuBvU0BTCCBDxUl8dGoxKmzkCnfXFlbOjD0zOxz3LafIo6AjN6HDAK+YQhRslhbt0uU3E5qpH0NMSUXh2WfVMiy2v6hcWIZ9QOqH4ojKbFzQqkuAIpMcVj8JWqc4NZPA5kjD0zDrgN/VppAQyD7glJxHGoNtwSZ8rqjdoCabkMYLHXHiRhBbUXTRJPKuCjxJIWGa2eCJnFkS1KWcap5obIs1q+tFnkPKsCGxWA8cWbLR0ixNcKbqHpjK8aZWQnqTD61FNegAEtqgtmnSVGhnJ2ZRRb9nhR2Wqh9wi6gNdHJWGKkhUA6rA6AS8Rp0ZzwauiAb0FtjntXA24pUmcw1KybAz//NetgAHAMGNVEo7haMqFnro83HQRQLH04qMPOvOII2daFjCVs9XN8N8kcXsRzJSYUH4AokQWlw5u1bG5Xq/r9bper+v1ul6v6/X6n16vrl1/7+Ln7/IX418=)

**Comment by MurakamiShinyu** [comment]
Antenna House and Prince's page floats are different from our current draft. AH/Prince's top/bottom floats are 1-dimensional, i.e., inline alignment of block-top/end floats are like normal blocks, and not floating in inline direction.

**Comment by johanneswilm** [comment]
@MurakamiShinyu Yes, but if I recall the research from back then correctly, they had issues dealing with several floats on the same page. Has this been resolved since?

**Comment by johanneswilm** [comment]
The proposed change si ok though, isn't it, @MurakamiShinyu ? I think the main reason we combined block-start/inline-start and block-end/inline-end was for reasons of symmetry. But this ends up being wierd: we allow floating only into two of four corners. So if we want to have true two-dimensional floats, we need to find the time and make the changed proposed a year ago.

But as long as we only support two corners, it can just as well be the block-end/line-start corner instead of the block-end/line-end corner. Most use-cases will probably anyway center around floats that fill the entire line-width if users cannot float into all corners (and stack into all directions).

**Comment by YoshithaRathnayake** [comment]
Shall I work on this ?
